cloud/kyup/kyup.py

```diff
@@ -0,0 +1,357 @@
+      - Storage size in GB. Minimum 20 GB.
+    required: false
+    default: 20
+  cpu_cores:
+    description:
+      - Number of CPU cores to be used. Minimum 2 CPU cores.
+    required: false
+    default: 2
+  bw:
+    description:
+      - Bandwidth limit for this container, in TB. Minimum 2 TB.
+    required: false
+    default: 2
+  storage_type:
+    description:
+      - Choose the storage type for this container.
+    required: false
+    default: local
+'''
+
+EXAMPLES = '''
+# Basic provisioning example
+---
+- hosts: kyup
+  tasks:
+  - name: ensure kyup container
+    kyup: >
+      api_key=XXXX
+      enc_key=YYYY
+      action=create
+      image="Centos Plain"
+      dc_id=1
+      name=avalon
+      password=e4a17265b28f6a9e8657
+      ssh_keys=hackman,piele
+  - add_host: name={{ container_ip }} group=containers
+
+- hosts: containers
+  tasks:
+  - name: copy the hosts file
+    copy: src=/etc/hosts dest=/etc/hosts
+
+
+# Rebooting a container
+---
+- hosts: kyup
+  tasks:
+  - name: restart the container
+    kyup: >
+      api_key=XXXX
+      action=restart
+      name=avalon
+
+'''
+
+RETURN = '''
+kyup_container_name:
+    returned: success
+    type: string
+    sample: "container_test"
+kyup_container_ip:
+    returned: success
+    type: string
+    sample: "10.12.131.5"
+'''
+
+import os
+import time
+import base64
+import md5
+from Crypto.Cipher import AES
+
+# import module snippets
+from ansible.module_utils.basic import *
+from ansible.module_utils.urls import *
+
+## Defaults for the module
+# API URL
+url='https://api.kyup.com/client/v1'
+
+# Modify this counter if you want to wait for the requests, more then a minute
+retries=12 # 12 x 5sec = 60 seconds
+
+
+## Code used for encryption of the passwords
+blocksize = AES.block_size
+
+def pad( data ):
+    padding = blocksize - ( len(data) % blocksize )
+    return data + (chr(padding) * padding)
+
+def encrypt( key, data ):
+    data = pad(data)
+    m = md5.new(key)
+    iv = m.hexdigest()[0:blocksize]
+    cipher = AES.new( key, AES.MODE_CBC, iv )
+    return base64.b64encode( cipher.encrypt( data ) )
+## End of encryption code
+
+def api_request(module, data):
+    ret = {}
+    http_status = 0
+    params = 'request=' + data
+    count = retries
+    while count:
+        # get the response
+        resp, info = fetch_url(module, url, params)
+        # parse the json
+        ret = module.from_json(resp.read())
+        status = ret.get('status', False)
+        if status:
+            return ret
+        http_status = info['status']
+        count -= 1
+        time.sleep(5)
+
+    if http_status != 200:
+        module.fail_json(changed=False, msg = 'Request failed with status %d' % int(status))
+
+    status = ret.get('status', False)
+    # Exit if we did not found status
+    if not status:
+        module.fail_json(changed=False, msg = 'Req: ' + data + ' Resp: ' + ret)
+
+    # We have found status, check its value here
+    if status:
```

Comment:
I do not agree. The status variable can be any of the following values:
0, 1, 101, 102 and so on.
The first check is if the status variable has assigned value.
The second check is if the value is true, if not do something else.